tools/d2g/dockerimageartifact.go

```diff
@@ -38,5 +28,5 @@ func (dia *DockerImageArtifact) FileMounts() ([]genericworker.FileMount, error)
 }
 
 func (dia *DockerImageArtifact) String() (string, error) {
-	return `"${IMAGE_NAME}"`, nil
+	return fmt.Sprintf("docker-archive:%s", filepath.Base(dia.Path)), nil
```

Comment:
```suggestion
	return "docker-archive:dockerimage"
```

Elsewhere, rather than attempting to preserve file names, which indeed might clash, we have chosen to used fixed names, chosen by d2g. For example artifacts get the names `artifact0`, `artifact1`, .... These names cannot conflict, because they are named by d2g, and the file names in the original tasks have no bearing over the file names created on the host, only in the container. This works since d2g chooses names which do not conflict with each other, but can break if a file can be created on the host where the name is arbitrary and chosen by the task.

Reply:
Nice, thanks for the great idea Pete!